core/domain/user_services.py

```diff
@@ -933,22 +976,270 @@ def create_new_user(gae_id, email):
+    user_details_change_list.
+
+    Args:
+        gae_id: str. The GAE ID of the full (parent) user trying to create new
+            profiles.
+        user_details_change_list: list(UserDetailsChange). The list of
+            user details changes to use for creation of new profiles.
+
+    Returns:
+        list(UserSettings). List of UserSettings objects created for the new
+        users.
+
+    Raises:
+        Exception. If the pin for parent user trying to create a new profile
+            is not set.
+        ValidationError. If a None or empty value is provided for display
+            alias attribute.
+    """
+
+    def _create_new_profiles_transactional(
+            user_settings_list, user_auth_details_list):
+        """Save user models for new users as a transaction.
+
+        Args:
+            user_settings_list: list(UserSettings). The list of user settings
+                domain objects corresponding to the newly created list of users.
+            user_auth_details_list: list(UserAuthDetails). The list of user auth
+                details domain object corresponding to the newly created list
+                of users.
+        """
+        _save_multiple_users_auth_details(user_auth_details_list)
+        _save_multiple_users_settings(user_settings_list)
+
+    # New profile creation is done by full (parent) user.
+    parent_user_auth_details = get_auth_details_by_gae_id(gae_id, strict=True)
+    if parent_user_auth_details.pin is None:
+        raise Exception(
+            'Pin must be set for a full user before creating a profile.')
+    parent_user_id = parent_user_auth_details.user_id
+
+    user_settings_list = []
+    user_auth_details_list = []
+
+
+    received_display_aliases = []
+    for user_details_change in user_details_change_list:
+        if (not user_details_change.display_alias or
+                not isinstance(
+                    user_details_change.display_alias, python_utils.BASESTRING)
+           ):
+            raise utils.ValidationError(
+                'Expected display alias to be a string, received %s' %
+                user_details_change.display_alias
+            )
+        received_display_aliases.append(user_details_change.display_alias)
+
+    if len(received_display_aliases) != len(set(received_display_aliases)):
+        raise Exception('New users have duplicate display aliases.')
+
+    all_profiles_auth_details = get_all_profiles_auth_details_by_parent_user_id(
+        parent_user_id)
+    all_user_ids = [model.user_id for model in all_profiles_auth_details]
+    all_user_ids.append(parent_user_id)
+    existing_display_aliases = [
+        user.display_alias for user in get_users_settings(all_user_ids)
+        if user.display_alias is not None
+    ]
+
+    if set(received_display_aliases).intersection(
+            set(existing_display_aliases)):
+        raise Exception('New users have duplicate display aliases.')
```

Comment:
@vinitamurthi @seanlip If we have a new profile creation request, if it sends a duplicate `display_alias` I am raising an exception. So None of the users would be created. Is this the expected behaviour or do we want to still populate the rest of the profiles who do not have any conflict and send back a message saying these profiles were backed up.

Normally since the data flow is one-way only we wouldn't have to deal with this problem if the android app itself disallows this behaviour. But if at all there is a possibility that duplicate `display_alias` can be sent (like a case that I mentioned where device A was primary, then it became non-primary and created additional profiles, and then it again became primary) --I want to follow above suggested approach. Does that seem okay? Let me know your thoughts.

Reply:
Removed the exception now - given our current scenario of syncing profiles, we are only using backend for making a backup of the local data. So we can avoid raising this exception.